packages/forms/src/UIForm/fieldsets/Fieldset/Fieldset.component.js

```diff
@@ -9,7 +9,7 @@ export default function Fieldset(props) {
 	const { title, items, options } = schema;
 
 	const widgets = items
-		.filter((itemSchema, index) => shouldRender(itemSchema.condition, props.properties, index))
+		.filter(itemSchema => shouldRender(itemSchema.condition, props.properties, itemSchema.key))
 		.map((itemSchema, index) => <Widget {...restProps} key={index} schema={itemSchema} />);
```

Comment:
could we use the `itemSchema.key` as the `key` for <Widget/> as well ?
This way we avoid the anti-patern of using and index as a key.